### ISSUE TITLE
Use One Hot Encoding for categorical creation

### ISSUE THREAD

**Issue by derekflint** [deeplearninc/auger-ml#40]
It looks like using StringIndexer is not the correct way to send categorical features to the ml algs.

See this article: https://hackernoon.com/what-is-one-hot-encoding-why-and-when-do-you-have-to-use-it-e3c6186d008f

Add one more step to take StringIndexer values and transform to one hot encoding.

https://spark.apache.org/docs/2.1.0/api/python/pyspark.ml.html#pyspark.ml.feature.OneHotEncoder

For labels we should use LabelBinarizer:
http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html

**Comment by khizanov** [comment]
https://github.com/deeplearninc/auger-ml/pull/94
